drivers/storage/portworx/portworx.go

```diff
@@ -893,6 +895,12 @@ func hasPrometheusChanged(cluster *corev1.StorageCluster) bool {
 		cluster.Status.DesiredImages.PrometheusOperator == ""
 }
 
+func hasPrometheusVersionChanged(cluster *corev1.StorageCluster, portworx *portworx) bool {
+	return portworx.k8sVersion != nil &&
```

Comment:
nit: Here we can assume k8sVersion is not nil as we return error from the very beginning if there is an error getting the k8sVersion.